packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -223,10 +230,18 @@ abstract class SerializationManager {
     bool formatted = false,
   }) {
     if (object is ProtocolSerialization) {
-      return encode(object.toJsonForProtocol(), formatted: formatted);
+      return encode(
+        object.toJsonForProtocol(),
```

Comment:
Question: Should we add a small comment here explaining that we do this because then we can skip the branching inside the `encode` method?